frontend/src/lib/components/apps/components/helpers/RunnableWrapper.svelte

```diff
@@ -64,6 +66,14 @@
 		wrapperClass={runnableClass}
 		wrapperStyle={runnableStyle}
 		{render}
+		on:success={() => {
+			if (Array.isArray(setTab)) {
+				setTab.forEach((tab) => {
+					const { id, index } = tab
+					$componentControl[id].setTab?.(index)
+				})
+			}
```

Comment:
let's handle the goto and gotoNewTab here

Reply:
Done! I've also moved the recompute code here as well